core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -108,8 +109,15 @@ public KafkaStore(SchemaRegistryConfig config,
         JaasUtils.isZkSecurityEnabled());
     this.brokerSeq = zkUtils.getAllBrokersInCluster();
 
-    this.bootstrapBrokers = KafkaStore.getBrokerEndpoints(
-            JavaConversions.seqAsJavaList(this.brokerSeq));
+    List<String> bootstrapServersConfig = config.getList(SchemaRegistryConfig.KAFKASTORE_BOOTSTRAP_SERVERS_CONFIG);
+    List<String> endpoints;
+    if (bootstrapServersConfig.isEmpty()) {
+      endpoints = brokersToEndpoints(JavaConversions.seqAsJavaList(this.brokerSeq));
+    } else {
+      endpoints = config.getList(SchemaRegistryConfig.KAFKASTORE_BOOTSTRAP_SERVERS_CONFIG);
+    }
+    this.bootstrapBrokers = filterBrokerEndpoints(endpoints);
```

Comment:
Not sure how important it is, but one thing you lose by making these both use a simple filter is that if you specify `kafkastore.bootstrap.servers`, you probably want it to be an error if you specify a URL type not supported by the schema registry, whereas if you are populating directly from ZK, that simple filtering is the right thing to do.

Being able to tell between these two scenarios could help users (e.g. they specified a SASL URL, but the schema registry doesn't support that). Then again, I think other PRs are about to bring in support for all the known URL types, so up to you whether you think it'll make sense to warn them of the issue.